Sources/BitcoinKit/Messages/HeadersMessage.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  HeadersMessage.swift
+//
+//  Copyright © 2018 Kishikawa Katsumi
+//  Copyright © 2018 BitcoinKit developers
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+public struct HeadersMessage {
+    // The main client will never send us more than this number of headers.
+    public static let MAX_HEADERS: Int = 2000
+
+    /// Number of block headers
+    public let count: VarInt
+    /// Block headers
+    public let headers: [BlockMessage]
+
+    public func serialized() -> Data {
+        var data = Data()
+        data += count.serialized()
+        for header in headers {
+            data += header.serialized()
+        }
+        return data
+    }
+
+    public static func deserialize(_ data: Data) throws -> HeadersMessage {
```

Comment:
Feels strange because others Messages' `deserialize()` are not `throws`.
But you guess others should also be `throws`?

Reply:
Yes.
Some of protocol messages need to be `throws`.
Refer to protocol message implementations of bitcoinj.